compliance/collection/main.go

```diff
@@ -37,6 +41,31 @@ var (
 	once sync.Once
 )
 
+// These options control behaviour of all node inventory related functions
+type cachedScanOpts struct {
+	nodeName            string
+	scanner             nodeinventorizer.NodeInventorizer
+	inventoryCachePath  string
+	backoffWaitCallback func(int64) int64
+}
+
+// waitAndIncreaseBackoff returns the duration the backoff should wait next time
+func waitAndIncreaseBackoff(waitTimeSeconds int64) int64 {
```

Comment:
This function mixes two concerns: calculation of the _next_ backoff and waiting for the amount of _current_ backoff. As such it should be better split in two functions because multiple concerns (another name - side effects) are problematic in code maintenance and testing.

Reply:
Done!